tools/clang/unittests/HLSL/ValidationTest.cpp

```diff
@@ -4207,6 +4211,127 @@ TEST_F(ValidationTest, ValidateWithHash) {
+// works as intended.
+
+TEST_F(ValidationTest, UnitTestExtValidationSupport) {
+  DxcDllExtValidationSupport m_dllExtSupport1;
+  DxcDllExtValidationSupport m_dllExtSupport2;
+
+  // capture any existing value in the environment variable,
+  // so that it can be restored after the test
+  std::wstring oldEnvVal = GetEnvVarW(L"DXC_DXIL_DLL_PATH");
+
+  // 1. with no env var set, test GetDxilDllPath() and DxilDllFailedToLoad()
+
+  // make sure the variable is cleared, in case other tests may have set it
+  SetEnvVarW(L"DXC_DXIL_DLL_PATH", L"");
+
+  // empty initialization should succeed
+  VERIFY_SUCCEEDED(m_dllExtSupport1.Initialize());
+
+  VERIFY_IS_FALSE(m_dllExtSupport1.DxilDllFailedToLoad());
+  VERIFY_ARE_EQUAL(m_dllExtSupport1.GetDxilDllPath(), "");
+
+  // 2. Test with a bogus path in the environment variable
+  SetEnvVarW(L"DXC_DXIL_DLL_PATH", L"bogus");
+
+  if (!m_dllExtSupport2.IsEnabled()) {
+    VERIFY_SUCCEEDED(m_dllExtSupport2.Initialize());
+  }
+
+  // validate that m_dllExtSupport2 was able to capture the environment
+  // variable's value, and that loading the bogus path was unsuccessful
+  std::string extPath("bogus");
+  VERIFY_ARE_EQUAL(m_dllExtSupport2.GetDxilDllPath(), extPath);
+  VERIFY_IS_TRUE(m_dllExtSupport2.DxilDllFailedToLoad());
+
+  // 3. Test production of class IDs CLSID_DxcCompiler, CLSID_DxcLinker,
+  // and CLSID_DxcValidator through DxcDllExtValidationSupport.
+  CComPtr<IDxcCompiler> pCompiler;
+  CComPtr<IDxcLinker> pLinker;
+  CComPtr<IDxcValidator> pValidator;
+  CComPtr<IDxcOperationResult> pResult;
+
+  VERIFY_SUCCEEDED(m_dllExtSupport2.CreateInstance(
+      CLSID_DxcCompiler, __uuidof(IDxcCompiler), (IUnknown **)&pCompiler));
+  VERIFY_SUCCEEDED(m_dllExtSupport2.CreateInstance(
+      CLSID_DxcLinker, __uuidof(IDxcLinker), (IUnknown **)&pLinker));
+  VERIFY_SUCCEEDED(m_dllExtSupport2.CreateInstance(
+      CLSID_DxcValidator, __uuidof(IDxcValidator), (IUnknown **)&pValidator));
+  CComPtr<IMalloc> pMalloc;
+  CComPtr<IDxcCompiler2> pCompiler2;
+  pLinker.Release();
+  pValidator.Release();
+  VERIFY_SUCCEEDED(DxcCoGetMalloc(1, &pMalloc));
+  VERIFY_SUCCEEDED(m_dllExtSupport2.CreateInstance2(pMalloc, CLSID_DxcCompiler,
+                                                    __uuidof(IDxcCompiler),
+                                                    (IUnknown **)&pCompiler2));
+  VERIFY_SUCCEEDED(m_dllExtSupport2.CreateInstance2(
+      pMalloc, CLSID_DxcLinker, __uuidof(IDxcLinker), (IUnknown **)&pLinker));
+  VERIFY_SUCCEEDED(m_dllExtSupport2.CreateInstance2(pMalloc, CLSID_DxcValidator,
+                                                    __uuidof(IDxcValidator),
+                                                    (IUnknown **)&pValidator));
+
+  // reset the environment variable to its previous value, if it had one.
+  if (!oldEnvVal.empty()) {
+    SetEnvVarW(L"DXC_DXIL_DLL_PATH", oldEnvVal);
+  } else {
+    ClearEnvVarW(L"DXC_DXIL_DLL_PATH");
+  }
```

Comment:
You can't differentiate unset from empty anyway, so just allow it to set to empty if it is.
```suggestion
 SetEnvVarW(L"DXC_DXIL_DLL_PATH", OldEnvVal);
```